comp/core/autodiscovery/common/utils/annotations.go

```diff
@@ -287,7 +292,14 @@ func extractTemplatesFromMapWithV2(entityName string, annotations map[string]str
 	}
 
 	if actualPrefix != "" {
-		c, err := extractLogsTemplatesFromMap(entityName, annotations, actualPrefix)
+		logCheckName := ""
+		if len(configs) >= 1 {
+			// Consider the first check name as the log check name, even if it's empty
+			logCheckName = configs[0].Name
+		}
```

Comment:
Would it be possible to make this code part of the `extractLogsTemplatesFromMap` instead of repeating it twice?

Reply:
I'm not sure we want to add `configs` as a parameter of `extractLogsTemplatesFromMap` ; computing it outside of the function seems better, and we might change the logic to compute the logCheckName. What do you think?

Reply:
Why not? Are we going to change the logic differently between the two codepaths?

Reply:
I'm adding it in the parameter since we're fine with it then 👍